### PR TITLE
Minor ASAN improvements

### PR DESCRIPTION
**Pull request by maleadt**
- Warn if our SIGSEGV handler will be blocked, see https://github.com/JuliaLang/julia/issues/17706#issuecomment-236300600
- Use larger signal stack, see https://github.com/JuliaLang/julia/issues/17706#issuecomment-236369266
- Use a single `JL_ASAN_ENABLED` option for all conditional code

### CODE REVIEW

**Comment by yuyichao** [comment]
Sigstack size change is covered by https://github.com/JuliaLang/julia/pull/17727

File: src/julia_internal.h [hunk 1]
```diff
@@ -12,6 +12,18 @@
 #define sleep(x) Sleep(1000*x)
 #endif
 
+#ifndef __has_feature
+#define __has_feature(x) 0
```
Comment (tkelman) [inline src/julia_internal.h]: defining this might change other code's behavior

Comment (maleadt) [inline src/julia_internal.h]: I got this from the [Clang documentation on this extension](http://clang.llvm.org/docs/LanguageExtensions.html#feature-checking-macros)

**Comment by maleadt** [comment]
Updated: removed the signal stack size change and removed the definition of `__has_feature`.

**Pull request merged**